extensions/amp-analytics/0.1/amp-analytics.js

```diff
@@ -240,6 +240,14 @@ export class AmpAnalytics extends AMP.BaseElement {
     return this.iniPromise_;
   }
 
+  /**
+   * @return {boolean} whether parent post messages are allowed.
+   * @private
+   */
+  allowParentPostMessage_() {
+    return this.isInabox_ || isInFie(this.element);
```

Comment:
The `isInFie` check is expensive. Let's cache this to a property.

Reply:
done.